src/main/java/seedu/address/model/comparator/NameComparator.java

```diff
@@ -0,0 +1,35 @@
+package seedu.address.model.comparator;
+import java.util.Comparator;
+
+import seedu.address.model.person.Person;
+
+
+public class NameComparator implements Comparator<Person> {
+    private final String comparator = "name";
+
+    @Override
+    public int compare(Person p1, Person p2) {
+        if (p1.getName() == null) {
+            if (p2.getName() == null) {
+                return 0;
+            }
+            return 1;
+        }
+        if (p2.getName() == null) {
+            return -1;
+        }
+        return p1.getName().fullName.compareTo(p2.getName().fullName);
```

Comment:
Would it be better if we use `p1.getName().toString()` since it also returns `fullName`, rather than accessing the variable directly. 

Or more generally would it be better if we create getter methods for these cases instead of accessing the variables directly?

Reply:
`p1.getName().fullName` was a method i followed based on its original code base, i think the app uses it sometimes so I just follow to be consistent, but yep toString() works too :))